cfme/tests/openstack/infrastructure/test_roles.py

```diff
@@ -1,39 +1,85 @@
 import pytest
-import re
+from random import choice
 
 from cfme.fixtures import pytest_selenium as sel
 from cfme.infrastructure.deployment_roles import DeploymentRoles
+from cfme.infrastructure.host import Host
 from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
-from cfme.web_ui import InfoBlock, Quadicon
+from cfme.web_ui import flash, Quadicon, summary_title
 from utils import testgen
+from utils.wait import wait_for
 from utils.appliance.implementations.ui import navigate_to
-from utils.version import current_version
 
 
 pytest_generate_tests = testgen.generate([OpenstackInfraProvider],
                                          scope='module')
-pytestmark = [pytest.mark.uncollectif(lambda: current_version() < '5.7'),
-              pytest.mark.usefixtures("setup_provider_modscope")]
+pytestmark = [pytest.mark.usefixtures("setup_provider_modscope")]
 
 
 ROLES = ['NovaCompute', 'Controller', 'Compute', 'BlockStorage', 'SwiftStorage',
          'CephStorage']
 
 
-def test_host_role_type(provider):
-    navigate_to(provider, 'Details')
-    sel.click(InfoBlock.element("Relationships", "Nodes"))
-    my_quads = list(Quadicon.all())
-    assert len(my_quads) > 0
-    for quad in my_quads:
-        role_name = str(quad.name)
-        role_name = re.search(r'\((\w+)\)', role_name).group(1)
-        assert role_name in ROLES
+def test_host_role_association(provider, soft_assert):
+    navigate_to(Host, 'All')
+    names = [q.name for q in list(Quadicon.all())]
+    for node in names:
+        host = Host(node, provider)
+        navigate_to(host, 'Details')
+        role_name = summary_title().split()[1].translate(None, '()')
+        role_name = 'Compute' if role_name == 'NovaCompute' else role_name
+        role_assoc = host.get_detail('Relationships', 'Deployment Role')
+        soft_assert(role_name in role_assoc, 'Deployment roles misconfigured')
```

Comment:
Again, I want to assert it for each node in a loop. It will help to debug failures also, as for me :)